File: Dataset/Leetcode/train/94/715.java
 class Solution {
    public List<Integer> XXX(TreeNode root) {
        ArrayList<Integer> res = new ArrayList<>();
        inOrder(res, root);
        return res;
    }
    private void inOrder(ArrayList<Integer> list, TreeNode node) {
        if (node == null) {
            return;
        }
        inOrder(list, node.left);
        list.add(node.val);
        inOrder(list, node.right);
    }
}

File: Dataset/Leetcode/train/110/715.js
 var XXX = function(root) {
  if (!root) return true
  return Math.abs(maxDepth(root.left) - maxDepth(root.right)) <= 1 && XXX(root.left) && XXX(root.right)
  function maxDepth(node) {
    if (!node) return 0
    return Math.max(maxDepth(node.left), maxDepth(node.right)) + 1
  }
}

